service/s3/errors.go

```diff
@@ -33,6 +33,12 @@ const (
 	// The specified bucket does not exist.
 	ErrCodeNoSuchBucket = "NoSuchBucket"
 
+	// ErrCodeBucketNotFound for service response error code
+	// "NotFound".
+	//
+	// The specified bucket does not exist
+	ErrCodeBucketNotFound = "NotFound"
```

Comment:
`NotFound` is a generic error code derived from the HTTP response message's status code, and can be returned for any S3 operation that responds with a 404 status code and no other error code present.  Due to this the constant `ErrCodeBucketNotFound`.

In addition, these constants are generated based on the API model defined by Amazon S3, modifications to the file directly will be lost during regeneration of the API.

With that said, we are looking at ways to work with the Amazon S3 team to better represent this error in a way that is more easily consumed.